app/lib/features/events/pages/event_list_page.dart

```diff
@@ -0,0 +1,224 @@
+import 'dart:math';
+
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/providers/space_providers.dart';
+import 'package:acter/common/toolkit/buttons/primary_action_button.dart';
+import 'package:acter/common/utils/routes.dart';
+import 'package:acter/common/widgets/add_button_with_can_permission.dart';
+import 'package:acter/common/widgets/empty_state_widget.dart';
+import 'package:acter/features/events/providers/event_providers.dart';
+import 'package:acter/features/events/widgets/event_item.dart';
+import 'package:acter/features/events/widgets/skeletons/event_list_skeleton_widget.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_staggered_grid_view/flutter_staggered_grid_view.dart';
+import 'package:go_router/go_router.dart';
+import 'package:acter/features/search/providers/search.dart';
+
+class EventListPage extends ConsumerStatefulWidget {
+  final String? spaceId;
+
+  const EventListPage({super.key, this.spaceId});
+
+  @override
+  ConsumerState<EventListPage> createState() => _EventListPageState();
+}
+
+class _EventListPageState extends ConsumerState<EventListPage> {
+  final TextEditingController searchTextController = TextEditingController();
+
+  String get searchValue => ref.watch(searchValueProvider);
```

Comment:
I can think this approach provides more clean ```build()``` but help me understand if this is as effective as deciding at ```build``` where ```.watch()``` is usually intended.

Reply:
This to make getter method of provider values and make it shorter. Also used in many places of the class so used in this way.

Reply:
hmmm... I see the appeal of doing it this way, but @gtalha07 is right in that this has the danger to be wrongly used if not being called in a `build` ... but as long as that is only within `build`-functions, that should be fine.